__tests__/V5ClusterManagement.js

```diff
@@ -447,3 +449,127 @@ it('renders an error message if there was an error loading apps', () => {
   expect(queryByTestId('installed-apps')).toBeNull();
   expect(queryByTestId('no-apps-found')).toBeNull();
 });
+
+it('renders cluster labels', async () => {
+  const clusterDetailPath = RoutePath.createUsablePath(
+    OrganizationsRoutes.Clusters.Detail.Home,
+    {
+      orgId: ORGANIZATION,
+      clusterId: V5_CLUSTER.id,
+    }
+  );
+  const { findByText, getByText, queryByText } = renderRouteWithStore(
+    clusterDetailPath
+  );
+
+  await findByText('Labels:');
+
+  for (const [key, value] of Object.entries(v5ClusterResponse.labels)) {
+    if (
+      key.includes(Constants.RESTRICTED_CLUSTER_LABEL_KEY_SUBSTRING) === false
+    ) {
+      expect(getByText(key)).toBeInTheDocument();
+      expect(getByText(value)).toBeInTheDocument();
+    } else {
+      expect(queryByText(key)).not.toBeInTheDocument();
+      expect(queryByText(value)).not.toBeInTheDocument();
+    }
+  }
+});
+
+it('allows to delete cluster labels', async () => {
+  mockCall('put', `/v5/clusters/${V5_CLUSTER.id}/labels/`, {
+    labels: {
+      'giantswarm.io/hidden-label': 'ok',
+    },
+  });
+  const clusterDetailPath = RoutePath.createUsablePath(
+    OrganizationsRoutes.Clusters.Detail.Home,
+    {
+      orgId: ORGANIZATION,
+      clusterId: V5_CLUSTER.id,
+    }
+  );
+  const { findByText, findByTestId } = renderRouteWithStore(clusterDetailPath);
+
+  await findByText('Labels:');
+
+  const visibleLabels = Object.entries(filterLabels(v5ClusterResponse.labels));
+
+  fireEvent.click(await findByTestId(`delete-${visibleLabels[0][0]}`));
+  fireEvent.click(await findByTestId('delete-label-button'));
+
+  await findByText(/This cluster has no labels./);
+});
+
+it('disallows to add invalid cluster labels', async () => {
+  const clusterDetailPath = RoutePath.createUsablePath(
+    OrganizationsRoutes.Clusters.Detail.Home,
+    {
+      orgId: ORGANIZATION,
+      clusterId: V5_CLUSTER.id,
+    }
+  );
+  const { findByText, findByTestId } = renderRouteWithStore(clusterDetailPath);
+
+  await findByText('Labels:');
+  fireEvent.click(await findByTestId('add-label-button'));
+
+  const keyInput = await findByTestId('label-key-input');
+  const valueInput = await findByTestId('label-value-input');
+  const saveButton = await findByTestId('save-label-button');
```

Comment:
What do you think about using more accessible queries? We had a discussion about this at some point, I believe before you joined:

These are the rules we usually guide by:
https://testing-library.com/docs/guide-which-query

Also, I see that you're using a lot of async queries. Are all of those really necessary, or can we replace them with sync ones? They have a big impact on the duration of the tests